test/Microsoft.AspNet.Razor.Test/Parser/CSharp/CSharpStatementTest.cs

```diff
@@ -137,6 +137,157 @@ public void CatchClause()
                                ));
         }
 
+        public static TheoryData ExceptionFilterData
+        {
+            get
+            {
+                var factory = SpanFactory.CreateCsHtml();
+
+                // document, expectedStatement
+                return new TheoryData<string, StatementBlock>
+                {
+                    {
+                        "@try { someMethod(); } catch(Exception) when (true) { handleIO(); }",
```

Comment:
Does Razor currently have issues parsing an exception filter nested inside other C# block types? e.g.
``` c#
  @{
    try
    {
    }
    catch (Exception) when (Exception is ArgumentException)
    {
    }
  }
```

I doubt this is a problem. But either way, add one test using an exception filter in a statement block.